inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/reduce_gpu_b_fs_yx_fsv16.cl

```diff
@@ -0,0 +1,406 @@
+        #elif REDUCE_PROD_MODE
+            ACCUMULATOR_TYPE next = ACCUMULATOR_VAL_ONE;
+            acc *= intel_sub_group_shuffle_down(acc, next, 8);
+            acc *= intel_sub_group_shuffle_down(acc, next, 4);
+            acc *= intel_sub_group_shuffle_down(acc, next, 2);
+            acc *= intel_sub_group_shuffle_down(acc, next, 1);
+            acc  = intel_sub_group_shuffle(acc, 0);
+        #elif REDUCE_AND_MODE
+            acc = sub_group_all(acc);
+        #elif REDUCE_OR_MODE
+            acc = sub_group_any(acc);
+        #elif REDUCE_SUM_SQUARE_MODE || REDUCE_L2_MODE
+            acc = sub_group_reduce_add(acc);
+        #elif REDUCE_L1_MODE
+            acc = sub_group_reduce_add(acc);
+        #elif REDUCE_LOG_SUM_EXP_MODE
+            acc = sub_group_reduce_add(acc);
+        #endif
+    #endif
+
+    return acc;
+}
+
+inline FINAL_ACCUMULATOR_TYPE FUNC(final_reduce)(FINAL_ACCUMULATOR_TYPE acc) {
+    #if REDUCE_MEAN_MODE
+        acc /= DIVIDER;
+    #elif REDUCE_L2_MODE
+        acc = sqrt(acc);
+    #elif REDUCE_LOG_SUM_MODE || REDUCE_LOG_SUM_EXP_MODE
+        acc = log(acc);
+    #endif
+
+    return acc;
+}
+
+inline uint FUNC(calc_linear_offset)(uint b, uint f, uint y, uint x) {
+    uint index = b * COMMON_OUTPUT_SIZE_X * COMMON_OUTPUT_SIZE_Y * COMMON_OUTPUT_FEATURE_NUM +
+                 f * COMMON_OUTPUT_SIZE_X * COMMON_OUTPUT_SIZE_Y +
+                 y * COMMON_OUTPUT_SIZE_X +
+                 x;
+
+    return index;
+}
+
+__attribute__((intel_reqd_sub_group_size(SIMD)))
+KERNEL(reduce_fsv16)(
+    const __global INPUT0_TYPE* data,
+    __global OUTPUT_TYPE* output
+#if HAS_FUSED_OPS_DECLS
+    , FUSED_OPS_DECLS
+#endif
+)
+{
+    const uint xy   = (uint)get_global_id(1) * READ_OFFSET;
+    const uint x    = xy % ((COMMON_OUTPUT_SIZE_X + READ_OFFSET - 1)/READ_OFFSET * READ_OFFSET);
+    const uint y    = xy / ((COMMON_OUTPUT_SIZE_X + READ_OFFSET - 1)/READ_OFFSET * READ_OFFSET);
+    const uint bf   = (uint)get_global_id(2) * SIMD;
+    const uint b    = bf / ((COMMON_OUTPUT_FEATURE_NUM + SIMD - 1)/SIMD * SIMD);
+    const uint f    = bf % ((COMMON_OUTPUT_FEATURE_NUM + SIMD - 1)/SIMD * SIMD);
+
+#if REDUCE_BATCH && REDUCE_FEATURE && REDUCE_X && !KEEP_DIMS                            // BFX
+    const uint out_idx = OUTPUT_GET_INDEX(y, x, b, f);
+#elif REDUCE_BATCH && REDUCE_FEATURE && REDUCE_Y && !KEEP_DIMS                          // BFY
+    const uint out_idx = OUTPUT_GET_INDEX(x, b, f, y);
+#elif REDUCE_FEATURE && REDUCE_X && !KEEP_DIMS                                          // FX
+    const uint out_idx = OUTPUT_GET_INDEX(b, y, x, f);
+#elif REDUCE_BATCH && REDUCE_X && !KEEP_DIMS                                            // BX
+    const uint out_idx = OUTPUT_GET_INDEX(f + get_sub_group_local_id(), y, x, b);
+#elif REDUCE_BATCH && REDUCE_Y && !KEEP_DIMS                                            // BY
+    const uint out_idx = OUTPUT_GET_INDEX(f + get_sub_group_local_id(), x, b, y);
```

Comment:
Maybe I miss something, but shouldn't indices order be similar for BY and BX (and some other) cases? 
i.e. BY -> `... b, y`;  BX -> `..., b, x`, etc

Reply:
It doesn't matter from accuracy point of view, since in this cases last dimensions are 1. However from code consistency point of view you're absolutely right. It's corrected now.